components/brave_rewards/core/rewards_database.cc

```diff
@@ -324,6 +321,26 @@ mojom::DBCommandResponse::Status RewardsDatabase::Migrate(
   return mojom::DBCommandResponse::Status::RESPONSE_OK;
 }
 
+bool RewardsDatabase::ShouldCreateTables() {
+  if (!sql::MetaTable::DoesTableExist(&db_)) {
+    return true;
+  }
+
+  return GetTablesCount() <= 1;
+}
+
+int RewardsDatabase::GetTablesCount() {
+  sql::Statement statement(db_.GetUniqueStatement(
+      "SELECT COUNT(*) FROM sqlite_schema WHERE type='table'"));
+
+  int tables_count = 0;
+  if (statement.Step()) {
+    tables_count = statement.ColumnInt(0);
+  }
+
+  return tables_count;
+}
```

Comment:
This could have been done differently, but I wanted it to match exactly how Ads does it.